rule/var_naming.go

```diff
@@ -234,6 +226,10 @@ func (w *lintNames) check(id *ast.Ident, thing string) {
 	})
 }
 
+func isUpperUnderScore(s string) bool {
+	return !hasLowerCaseLetter(s) && strings.Contains(s, "_")
+}
```

Comment:
There is a problem here, now you moved to Unicode function you will face problems

Some unicode letter are neither upper or lower case, so not being lower don't make it upper case.

This is true for at least CJK and Arabic

But it's also true with digits

```suggestion
// isUpperUnderScore detects variable that are made from upper case letters, underscore, or digits.
//
// short variable names are considered OK
func isUpperUnderScore(s string) bool {
	if !strings.Contains(s, "_") {
		return false
	}

	if len(s) <= 5 {
		// avoid false positive
		return false
	}

	for _, r := range s { 
		if r == '_' || isUpperOrDigit(r) {
			continue
		}
		return false
	}
	return true
}
```

This change includes my suggestion to check the length in the method: https://github.com/mgechev/revive/pull/1322#discussion_r2054390869

If you discard it, you will have to check s is not empty or underscore